doc/develop/modules.rst

```diff
@@ -1148,7 +1148,26 @@ Where 23 in the example above indicated the pull request number submitted to the
 *my_module* repository. Once the module changes are reviewed and merged, the
 revision needs to be changed to the commit hash from the module repository.
 
+.. _optional_modules:
 
+Enabling optional modules
+=========================
+
+Some of the modules are marked as optional. In order to use such modules, they
+need to be explicitly enabled in local west configuration.
+To enable a module run:
+
+.. code-block:: console
+
+    west config manifest.project-filter -- +<module name>
```

Comment:
Terminology here is wrong, and thus can lead to confusion.

Zephyr has modules, and those modules are generally always optional.
It may be that if you build for a specific board, like Nordic Semiconductor's nRF52840dk, then you need the HAL Nordic module.

But if you build for a STM board, then you don't need HAL Nordic, instead you need HAL st/stm.

And Zephyr modules can be provided without the use of `west`, https://docs.zephyrproject.org/latest/develop/modules.html#integrate-modules-in-zephyr-build-system

What you are doing with the `west config manifest.project-filter` is to activate a `west` project.
A `west` project may also be a Zephyr module, but doesn't have to be, in fact there are projects in the manifest which are not Zephyr modules, such as `bsim`, `net-tools`.

What you're trying to describe here, is already covered by [West Manifest - Active and Inactive Projects](https://docs.zephyrproject.org/latest/develop/west/manifest.html#active-and-inactive-projects)